### DISCUSSION TITLE
Is it possible to change the name record type for TanStack Form Arrays?

### DISCUSSION THREAD

**Discussion by nihilum-labs** [TanStack/form#914] [Q&A] [answered]
Hello, Is it possible to change type **people[${i}].name** to **people.${i}.name**?
```
<form.Field key={i} name={`people[${i}].name`}>
  {(subField) => (
    <input
      value={subField.state.value}
      onChange={(e) =>
        subField.handleChange(e.target.value)
      }
    />
  )}
</form.Field>
```

**Comment by Balastrong** [comment, marked as answer]
The short answer is no, that's how the entire library and type system are built.

What is your use case that needs this syntax?

**Reply by nihilum-labs** [reply]
@crutchcorn recommended the formData parsing library [Decode](https://github.com/fabian-hiller/decode-formdata). Unfortunately, at the moment it only works with the .$. format

I will look for another option, maybe you will recommend something?

Thank you for answer.

**Reply by Balastrong** [reply]
We're using the same library on our server validator (nextjs/start) and I think you can try with something similar to:
```tsx
<form.Field key={i} name={`hobbies[${i}]`}>
  {(subField) => {
    return (
          <input
            name={`hobbies.${i}`}
...
```

But doesn't seem ideal, maybe @crutchcorn already had an idea on how to handle that :)
